foundation-ui/auth/AuthService.js

```diff
@@ -0,0 +1,105 @@
+import Hooks from '../hooks';
+
+const hooks = Hooks();
+
+/**
+ * AuthService relies on a single instance of Hooks.
+ * It utilses the hooks actions to notify AuthPoints of changes and
+ * filters to allow external packages to register for and change the content
+ * output in the AuthPoint
+ *
+ * @example
+ * User of service would listen for updates and possibly change it
+ * import {AuthService} from 'foundation-ui/mount';
+ * AuthService.register('myID', function (value) {
+ *   return value.replace('my', 'myChanged');
+ * });
+ *
+ * AuthPoint would register itself with the AuthService to be able to get
+ * updates on content
+ * import {AuthService} from 'foundation-ui/mount';
+ * function updateContent() {
+ *   let newContent = AuthService.getContent(
+ *     'myID',
+ *     React.Children.toArray(children),
+ *     props
+ *   );
+ *
+ *   // Do something with newContent
+ * }
+ * AuthService.addListener('myID', updateContent);
+ */
+const AuthService = {
+  // Package API
+  /**
+   * Registers package for updates on this AuthPoint permission
+   * @param  {String} permission of MointPoint to register package for
+   * @param  {Function} callback to call when content is being fetched
+   * @param  {Number} priority to register package callback with
+   * [-Infinity; Infinity]
+   */
+  register(permission, callback, priority) {
+    hooks.addFilter(permission, callback, priority);
+    // Call all actions registered for permission.
+    // This will call an action for all AuthPoints with this permission
+    // and have them update state by running a new filter
+    // (which will include the newly added filter)
+    hooks.doAction(permission);
+  },
+
+  /**
+   * Unregisters package for updates on this AuthPoint permission
+   * @param  {String} permission listener is registered with
+   * @param  {Function} callback registered with listener
+   */
+  unregister(permission, callback) {
+    hooks.removeFilter(permission, callback);
+    // Have AuthPoints run filter again so removed filter will be excluded
+    hooks.doAction(permission);
+  },
+
+  // AuthPoint API
+  /**
+   * Register for changes in AuthService specific to the given permission.
+   * Whenever AuthService has a change it invoke callback
+   * so we can get the newly filtered authorized value.
+   * @param {String} permission to listen for events
+   * @param {Function} callback to call when event fires
+   * @param {Number} priority of callback [-Infinity; Infinity]
+   */
+  addListener(permission, callback, priority) {
```

Comment:
Let's either use a different method name or emit a general "change" event to avoid confusion with common event emitter/dispatcher implementations.